qiling/os/posix/syscall/unistd.py

```diff
@@ -3,378 +3,379 @@
 
-    real_path = ql.os.path.transform_to_real_path(path)
-    relative_path = ql.os.path.transform_to_relative_path(path)
+    regreturn = 0 if os.path.exists(real_path) else -1
 
-    if os.path.exists(real_path) == False:
-        regreturn = -1
-    else:
-        regreturn = 0
+    # ql.log.debug("access(%s, 0x%x) = %d " % (relative_path, access_mode, regreturn))
 
-    ql.log.debug("access(%s, 0x%x) = %d " % (relative_path, access_mode, regreturn))
     if regreturn == 0:
-        ql.log.debug("File found: %s" % relative_path)
+        ql.log.debug(f'File found: {relative_path}')
     else:
-        ql.log.debug("No such file or directory")
+        ql.log.debug(f'No such file or directory: {relative_path}')
 
     return regreturn
 
 
-def ql_syscall_close(ql, close_fd, *args, **kw):
-    regreturn = -1
-    if 0 <= close_fd < NR_OPEN and ql.os.fd[close_fd] != 0:
-        ql.os.fd[close_fd].close()
-        ql.os.fd[close_fd] = 0
+def ql_syscall_close(ql: Qiling, fd: int):
+    if 0 <= fd < NR_OPEN and ql.os.fd[fd] != 0:
+        ql.os.fd[fd].close()
+        ql.os.fd[fd] = 0
         regreturn = 0
-    return regreturn
+    else:
+        regreturn = -1
 
+    return regreturn
 
-def ql_syscall_pread64(ql, read_fd, read_buf, read_len, read_offt, *args, **kw):
-    data = None
 
+def ql_syscall_pread64(ql: Qiling, fd: int, buf: int, length: int, offt: int):
     # https://chromium.googlesource.com/linux-syscall-support/+/2c73abf02fd8af961e38024882b9ce0df6b4d19b
     # https://chromiumcodereview.appspot.com/10910222
     if ql.archtype == QL_ARCH.MIPS:
-        read_offt = ql.unpack64(ql.mem.read(ql.reg.arch_sp + 0x10, size=0x08))
+        offt = ql.unpack64(ql.mem.read(ql.reg.arch_sp + 0x10, 8))
 
-    if 0 <= read_fd < NR_OPEN and ql.os.fd[read_fd] != 0:
+    if 0 <= fd < NR_OPEN and ql.os.fd[fd] != 0:
         try:
-            pos = ql.os.fd[read_fd].tell()
-            ql.os.fd[read_fd].lseek(read_offt)
-            data = ql.os.fd[read_fd].read(read_len)
-            ql.os.fd[read_fd].lseek(pos)
-            ql.mem.write(read_buf, data)
+            pos = ql.os.fd[fd].tell()
+            ql.os.fd[fd].lseek(offt)
+
+            data = ql.os.fd[fd].read(length)
+            ql.os.fd[fd].lseek(pos)
+
+            ql.mem.write(buf, data)
             regreturn = len(data)
         except:
             regreturn = -1
     else:
         regreturn = -1
+
     return regreturn
 
 
-def ql_syscall_read(ql, read_fd, read_buf, read_len, *args, **kw):
-    data = None
-    if 0 <= read_fd < NR_OPEN and ql.os.fd[read_fd] != 0:
+def ql_syscall_read(ql: Qiling, fd, buf: int, length: int):
+    if 0 <= fd < NR_OPEN and ql.os.fd[fd] != 0:
         try:
-            data = ql.os.fd[read_fd].read(read_len)
-            ql.mem.write(read_buf, data)
+            data = ql.os.fd[fd].read(length)
+            ql.mem.write(buf, data)
+
+            ql.log.debug(f'read() CONTENT: {data.decode()!r}')
```

Comment:
This line might cause below error
`UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe6 in position 25: invalid continuation byte`

Reply:
You are right; I already pushed another PR to fix it.